api/envoy/config/filter/http/header_to_metadata/v2/header_to_metadata.proto

```diff
@@ -20,6 +20,19 @@ message Config {
   enum ValueType {
     STRING = 0;
     NUMBER = 1;
+
+    // The value is a serialized `protobuf.Value
+    // <https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto#L62>`_.
+    PROTOBUF_VALUE = 2;
+  }
+
+  // ValueEncode defines the encoding algorithm.
+  enum ValueEncode {
+    // The value is not encoded.
+    NONE = 0;
+
+    // The value is encoded in `Base64Url <https://tools.ietf.org/html/rfc4648#section-5>`_.
+    BASE64URL = 1;
```

Comment:
Because there is only a limited set of characters available in HTTP headers, I think that ValueEncode = BASE64URL will always have to be set for protobuf metadata values obtained from headers. I also think that for configuration-specified metadata values, most users will find it easier to base64-encode the value. That leads me to think that we should just say that ValueType = PROTOBUF_VALUE indicates that value is a base64-encoded serialized protobuf and drop ValueEncode. This also nearly eliminates the odd case where ValueType = NUMBER and ValueEncode = BASE64URL means value is an integer string, encoded in base64.

Separately, can you confirm that the base64 encoding style referenced here is the same we use in other places?

Reply:
Sorry. I missed the earlier discussion about strings. It's fine top keep the encoding separate from the value type, but I think the integer encoding when base64 is enabled needs to be addressed. It's probably sufficient to just document it.

Reply:
Thanks, updated the comment.